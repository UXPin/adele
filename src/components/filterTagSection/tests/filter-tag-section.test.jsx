import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import FilterTagSection from '../filter-tag-section';

const testString = 'test-string';
const mockFunc = () => true;

it('renders FilterTagSection with 3 filters correctly', () => {
  const wrapper = mount(<FilterTagSection
    numberOfFilters={3}
    getFilters={mockFunc}
    clearFilters={mockFunc}
  />);
  expect(wrapper).toMatchSnapshot();
});


it('renders FilterTagSection with 0 filters correctly', () => {
  const wrapper = mount(<FilterTagSection
    numberOfFilters={0}
    getFilters={mockFunc}
    clearFilters={mockFunc}
  />);
  expect(wrapper).toMatchSnapshot();
});

it('renders FilterTagSection with 1 filters correctly', () => {
  const wrapper = mount(<FilterTagSection
    numberOfFilters={1}
    getFilters={mockFunc}
    clearFilters={mockFunc}
  />);
  expect(wrapper).toMatchSnapshot();
});
