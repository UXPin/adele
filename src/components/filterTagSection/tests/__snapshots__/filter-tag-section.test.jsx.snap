// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FilterTagSection with 0 filters correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 90%;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
}

<FilterTagSection
  clearFilters={[Function]}
  getFilters={[Function]}
  numberOfFilters={0}
>
  <styled.section>
    <section
      className="c0"
    />
  </styled.section>
</FilterTagSection>
`;

exports[`renders FilterTagSection with 1 filters correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 90%;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
}

<FilterTagSection
  clearFilters={[Function]}
  getFilters={[Function]}
  numberOfFilters={1}
>
  <styled.section>
    <section
      className="c0"
    />
  </styled.section>
</FilterTagSection>
`;

exports[`renders FilterTagSection with 3 filters correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 90%;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
}

.c1 {
  margin: 10px 0 10px 10px;
  color: #EF0A61;
  font-size: 10px;
  cursor: pointer;
  border: none;
  background: none;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<FilterTagSection
  clearFilters={[Function]}
  getFilters={[Function]}
  numberOfFilters={3}
>
  <styled.section>
    <section
      className="c0"
    >
      <styled.button
        onClick={[Function]}
        tabIndex={1}
      >
        <button
          className="c1"
          onClick={[Function]}
          tabIndex={1}
        >
          Remove All
        </button>
      </styled.button>
    </section>
  </styled.section>
</FilterTagSection>
`;
